test/extensions/filters/http/aws_lambda/config_test.cc

```diff
@@ -40,7 +40,7 @@ invocation_mode: asynchronous
   Http::MockFilterChainFactoryCallbacks filter_callbacks;
   auto has_expected_settings = [](std::shared_ptr<Envoy::Http::StreamFilter> stream_filter) {
     auto filter = std::static_pointer_cast<Filter>(stream_filter);
-    const auto settings = filter->settingsForTest();
+    const auto& settings = filter->settingsForTest();
```

Comment:
why to add `&` here?

Instead of making change in line 43, we can consider changing line 44 to `settings->payloadPassthrough()` ?

Reply:
setting is not copiable due to the unique_ptr that contains the signer, then taking the ref is legal in this scenario

Reply:
do you mean to return the setting shared_ptr?

Reply:
would this work?
```
auto filter = std::static_pointer_cast<Filter>(stream_filter);
    const auto settings = filter->settingsForTest();
    return settings->payloadPassthrough() &&
           settings->invocationMode() == InvocationMode::Asynchronous;
```

Reply:
the problem is that this

const auto settings = filter->settingsForTest();

is equivalent to 

const FilterSettings settings = filter->settingsForTest();

it try to create a concrete new object for settings, instead of getting the const FilterSettings& settings